src/libraries/Microsoft.PowerFx.Core/Public/Types/OptionSetValueType.cs

```diff
@@ -1,14 +1,15 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+using Microsoft.PowerFx.Core.Entities;
 using Microsoft.PowerFx.Core.Types;
 
 namespace Microsoft.PowerFx.Core.Public.Types
 {
     public class OptionSetValueType : FormulaType
     {
-        internal OptionSetValueType()
-            : base(new DType(DKind.OptionSetValue))
+        internal OptionSetValueType(IExternalOptionSet optionSet)
+            : base(DType.CreateOptionSetValue(optionSet))
```

Comment:
CreateOptionSetValueType(...) would be better, there are precedents (CreateNamedValueType)